recipe_recommender.mdx

```diff
@@ -0,0 +1,395 @@
+2. a list of recipes based on the query results from Pinecone
+
+<CodeGroup>
+```python suggestion.py
+import os
+
+import promptlayer
+
+openai = promptlayer.openai
+openai.api_key = os.getenv('OPENAI_API_KEY')
+
+def add_recipes_to_prompt(prompt, recipes_list):
+    """Combines the given prompt with the list of recipes to form a single string"""
+    recipes_string = '\n\n'.join(recipes_list)
+    results = prompt + '\n\n' + recipes_string
+    return results
+
+def suggest_final_recipe(prompt, recipes_list):
+    """Given a prompt and a list of recipes, use GPT to suggest one of the recipes"""
+    prompt_with_recipes = add_recipes_to_prompt(prompt, recipes_list)
+    response = openai.ChatCompletion.create(
+        model='gpt-3.5-turbo',
+        messages=[
+            {
+                'role': 'system',
+                'content': (
+                    'You are a helpful assistant. '
+                    'Given some user input and a list of recipes, suggest one of the recipes. '
+                    'The final output should be the full recipe.'
+                )
+            },
+            {'role': 'user', 'content': prompt_with_recipes},
+        ],
+        temperature=0.5,
+        max_tokens=1024
+    )
+    print(response.choices[0].message.content)
+```
+</CodeGroup>
+
+Lastly, create a `pinecone_query.py` file. In this file we provide a text snippet of food that we are interested
+in. Using that text, we will query our Pinecone index to find the nearest vector matches (closely related
+recipe IDs). Using those IDs, we can retrieve the original recipe text from our data. With that recipe text, we
+generate our final prompt which is sent to OpenAI, using the `suggest_final_recipe` function created in the previous
+step.
+
+<CodeGroup>
+```python pinecone_query.py
+import os
+
+import pinecone
+
+from files import read_txt_file
+from openai_utils import get_vector_for_text
+from settings import DATA_DIR, PINECONE_INDEX_NAME
+from suggestion import suggest_final_recipe
+
+PINECONE_API_KEY = os.environ.get('PINECONE_API_KEY')
+
+def get_recipes_from_ids(recipe_ids):
+    """Retrieve recipe contents for a given list of recipe IDs"""
+    results = []
+    filepaths = {DATA_DIR / f'{recipe_id}.txt' for recipe_id in recipe_ids}
+
+    for filepath in filepaths:
+        content = read_txt_file(filepath)
+        results.append(content)
+
+    return results
+
+def get_related_recipe_ids_for_text(text):
+    """Fetch IDs of recipes related to the given text using Pinecone"""
+    index = pinecone.Index(PINECONE_INDEX_NAME)
+    vector = get_vector_for_text(text)
+    query_results = index.query(
+        vector=vector,
+        top_k=2,
+        include_values=True
+    )
+    return {i['id'] for i in query_results['matches']}
+
+def main():
+    prompt = 'I like chicken.'
+    recipe_ids = get_related_recipe_ids_for_text(prompt)
+    recipes_list = get_recipes_from_ids(recipe_ids)
+    suggest_final_recipe(prompt, recipes_list)
```

Comment:
Not sure this order makes sense..

I got to this line excited to read the most important part... the suggestion! And was like wait.... did we already do `suggest_final_recipe`? Where is it?